com.hazelcast.internal.ascii.RestClusterMultiendpointTest

_master_ (commit 89254db9ee6767f998f897154d440ece6e5ae482)

Failed on Windows with OracleJDK8: http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Windows-OracleJDK8/318/testReport/com.hazelcast.internal.ascii/RestClusterMultiendpointTest/

Following tests failed:
- `testGetClusterState`
- `testChangeClusterState`

Stacktrace for `testGetClusterState`:
```
org.junit.ComparisonFailure: expected:<[pass]ive> but was:<[act]ive>
	at org.junit.Assert.assertEquals(Assert.java:117)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at com.hazelcast.internal.ascii.RestClusterTest.assertJsonContains(RestClusterTest.java:373)
	at com.hazelcast.internal.ascii.RestClusterTest.assertSuccessJson(RestClusterTest.java:382)
	at com.hazelcast.internal.ascii.RestClusterTest.testGetClusterState(RestClusterTest.java:152)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Stacktrace for `testGetClusterState`:
```
java.lang.AssertionError: Instance Member [127.0.0.1]:5035 - b26361e6-8c5c-4812-b5a4-5a9f7036bcef this expected:<FROZEN> but was:<ACTIVE>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterState(HazelcastTestSupport.java:1099)
	at com.hazelcast.test.HazelcastTestSupport$13.run(HazelcastTestSupport.java:1108)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1260)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1362)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterStateEventually(HazelcastTestSupport.java:1105)
	at com.hazelcast.internal.ascii.RestClusterTest.testChangeClusterState(RestClusterTest.java:167)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output for `testGetClusterState`:
```
12:48:00,700  WARN |testChangeClusterState| - [TcpServerConnection] hz.loving_blackwell.IO.thread-in-0 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] Connection[id=1, /0.0.0.0:51848->/127.0.0.1:5033, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5033, alive=false, connectionType=MEMBER, planeIndex=-1] closed. Reason: Exception in Connection[id=1, /0.0.0.0:51848->/127.0.0.1:5033, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5033, alive=true, connectionType=MEMBER, planeIndex=-1], thread=hz.loving_blackwell.IO.thread-in-0
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method) ~[?:1.8.0_172]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43) ~[?:1.8.0_172]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_172]
	at sun.nio.ch.IOUtil.read(IOUtil.java:197) ~[?:1.8.0_172]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[?:1.8.0_172]
	at com.hazelcast.internal.networking.nio.NioInboundPipeline.process(NioInboundPipeline.java:114) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKey(NioThread.java:383) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKeys(NioThread.java:368) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.selectLoop(NioThread.java:294) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.executeRun(NioThread.java:249) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
12:48:00,700  INFO |testChangeClusterState| - [TcpIpJoiner] hz.loving_blackwell.IO.thread-in-0 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5033 is added to the blacklist.
12:48:00,750  INFO |testChangeClusterState| - [TcpIpJoiner] hz.loving_blackwell.cached.thread-4 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5033 is removed from the blacklist.
12:48:00,750  INFO |testChangeClusterState| - [TcpServerConnection] hz.loving_blackwell.IO.thread-out-1 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:52112 and /127.0.0.1:5033
12:48:00,751  INFO |testChangeClusterState| - [TcpServerConnection] hz.reverent_blackwell.IO.thread-in-1 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5033 and /127.0.0.1:52112
12:48:00,751  WARN |testChangeClusterState| - [TcpServerConnection] hz.reverent_blackwell.IO.thread-in-1 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] Connection[id=2, /127.0.0.1:5033->/127.0.0.1:52112, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=null, alive=false, connectionType=MEMBER, planeIndex=-1] closed. Reason: Exception in Connection[id=2, /127.0.0.1:5033->/127.0.0.1:52112, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=null, alive=true, connectionType=MEMBER, planeIndex=-1], thread=hz.reverent_blackwell.IO.thread-in-1
java.lang.NullPointerException: null
	at com.hazelcast.internal.server.tcp.MemberProtocolEncoder.signalProtocolLoaded(MemberProtocolEncoder.java:101) ~[classes/:?]
	at com.hazelcast.internal.server.tcp.SingleProtocolDecoder.onRead(SingleProtocolDecoder.java:107) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioInboundPipeline.process(NioInboundPipeline.java:136) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioPipeline.lambda$start$0(NioPipeline.java:127) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processTaskQueue(NioThread.java:355) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.selectLoop(NioThread.java:290) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.executeRun(NioThread.java:249) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
12:48:00,901  WARN |testGetClusterState| - [TcpServerConnection] hz.unruffled_blackwell.IO.thread-in-1 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] Connection[id=2, /127.0.0.1:51984->/127.0.0.1:5039, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5039, alive=false, connectionType=MEMBER, planeIndex=-1] closed. Reason: Exception in Connection[id=2, /127.0.0.1:51984->/127.0.0.1:5039, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5039, alive=true, connectionType=MEMBER, planeIndex=-1], thread=hz.unruffled_blackwell.IO.thread-in-1
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method) ~[?:1.8.0_172]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43) ~[?:1.8.0_172]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_172]
	at sun.nio.ch.IOUtil.read(IOUtil.java:197) ~[?:1.8.0_172]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[?:1.8.0_172]
	at com.hazelcast.internal.networking.nio.NioInboundPipeline.process(NioInboundPipeline.java:114) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKey(NioThread.java:383) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKeys(NioThread.java:368) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.selectLoop(NioThread.java:294) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.executeRun(NioThread.java:249) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
12:48:00,901  INFO |testGetClusterState| - [TcpIpJoiner] hz.unruffled_blackwell.IO.thread-in-1 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5039 is added to the blacklist.
12:48:00,904  INFO |testGetClusterState| - [TcpIpJoiner] hz.unruffled_blackwell.cached.thread-6 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5039 is removed from the blacklist.
12:48:00,904  INFO |testGetClusterState| - [TcpServerConnection] hz.unruffled_blackwell.IO.thread-out-2 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:52113 and /127.0.0.1:5039
12:48:00,905  INFO |testGetClusterState| - [TcpServerConnection] hz.heuristic_blackwell.IO.thread-in-2 - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5039 and /127.0.0.1:52113
12:48:00,905  WARN |testGetClusterState| - [TcpServerConnection] hz.heuristic_blackwell.IO.thread-in-2 - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Connection[id=3, /127.0.0.1:5039->/127.0.0.1:52113, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=null, alive=false, connectionType=MEMBER, planeIndex=-1] closed. Reason: Exception in Connection[id=3, /127.0.0.1:5039->/127.0.0.1:52113, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=null, alive=true, connectionType=MEMBER, planeIndex=-1], thread=hz.heuristic_blackwell.IO.thread-in-2
java.lang.NullPointerException: null
	at com.hazelcast.internal.server.tcp.MemberProtocolEncoder.signalProtocolLoaded(MemberProtocolEncoder.java:101) ~[classes/:?]
	at com.hazelcast.internal.server.tcp.SingleProtocolDecoder.onRead(SingleProtocolDecoder.java:107) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioInboundPipeline.process(NioInboundPipeline.java:136) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioPipeline.lambda$start$0(NioPipeline.java:127) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processTaskQueue(NioThread.java:355) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.selectLoop(NioThread.java:290) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.executeRun(NioThread.java:249) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
12:48:01,051  WARN |testClusterShutdown| - [TcpServerConnection] hz.ecstatic_blackwell.IO.thread-in-1 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Connection[id=2, /127.0.0.1:52109->/127.0.0.1:5055, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5055, alive=false, connectionType=MEMBER, planeIndex=-1] closed. Reason: Exception in Connection[id=2, /127.0.0.1:52109->/127.0.0.1:5055, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5055, alive=true, connectionType=MEMBER, planeIndex=-1], thread=hz.ecstatic_blackwell.IO.thread-in-1
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method) ~[?:1.8.0_172]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43) ~[?:1.8.0_172]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_172]
	at sun.nio.ch.IOUtil.read(IOUtil.java:197) ~[?:1.8.0_172]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[?:1.8.0_172]
	at com.hazelcast.internal.networking.nio.NioInboundPipeline.process(NioInboundPipeline.java:114) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKey(NioThread.java:383) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKeys(NioThread.java:368) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.selectLoop(NioThread.java:294) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.executeRun(NioThread.java:249) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
12:48:01,051  INFO |testClusterShutdown| - [TcpIpJoiner] hz.ecstatic_blackwell.IO.thread-in-1 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] [127.0.0.1]:5055 is added to the blacklist.
12:48:01,053  INFO |testClusterShutdown| - [TcpIpJoiner] hz.ecstatic_blackwell.cached.thread-3 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] [127.0.0.1]:5055 is removed from the blacklist.
12:48:01,053  INFO |testClusterShutdown| - [TcpServerConnection] hz.ecstatic_blackwell.IO.thread-out-2 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:52114 and /127.0.0.1:5055
12:48:01,053  INFO |testClusterShutdown| - [TcpServerConnection] hz.laughing_blackwell.IO.thread-in-2 - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5055 and /127.0.0.1:52114
12:48:02,256  INFO |testClusterShutdown| - [ClusterService] hz.laughing_blackwell.priority-generic-operation.thread-0 - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5055 - 83c9d5a1-854d-4bc7-8f85-d202678a94be this
	Member [127.0.0.1]:5061 - 759052b1-9f04-46d8-9f76-204e98af6674
]

12:48:02,257  INFO |testClusterShutdown| - [ClusterService] hz.ecstatic_blackwell.priority-generic-operation.thread-0 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5055 - 83c9d5a1-854d-4bc7-8f85-d202678a94be
	Member [127.0.0.1]:5061 - 759052b1-9f04-46d8-9f76-204e98af6674 this
]

12:48:03,257  INFO |testClusterShutdown| - [LifecycleService] testClusterShutdown - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] [127.0.0.1]:5061 is STARTED
12:48:03,260  WARN |testClusterShutdown| - [NodeExtension] hz.ecstatic_blackwell.cached.thread-3 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Security features are only available on Hazelcast Enterprise!
12:48:03,261  INFO |testClusterShutdown| - [LockClusterStateOp] hz.ecstatic_blackwell.cached.thread-3 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Locking cluster state. Initiator: [127.0.0.1]:5061, lease-time: 60000
12:48:03,261  INFO |testClusterShutdown| - [LockClusterStateOp] hz.laughing_blackwell.generic-operation.thread-0 - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] Locking cluster state. Initiator: [127.0.0.1]:5061, lease-time: 60000
12:48:03,262  INFO |testClusterShutdown| - [LockClusterStateOp] hz.ecstatic_blackwell.cached.thread-3 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Extending cluster state lock. Initiator: [127.0.0.1]:5061, lease-time: 20000
12:48:03,262  INFO |testClusterShutdown| - [LockClusterStateOp] hz.laughing_blackwell.generic-operation.thread-0 - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] Extending cluster state lock. Initiator: [127.0.0.1]:5061, lease-time: 20000
12:48:03,263  INFO |testClusterShutdown| - [CommitClusterStateOp] hz.ecstatic_blackwell.cached.thread-3 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Changing cluster state from ClusterState{state=ACTIVE, lock=LockGuard{lockOwner=[127.0.0.1]:5061, lockOwnerId='46628225-c32e-4054-9db8-9d4c8f184db8', lockExpiryTime=1630759763261}} to ClusterStateChange{type=class com.hazelcast.cluster.ClusterState, newState=PASSIVE}, initiator: [127.0.0.1]:5061, transient: true
12:48:03,263  INFO |testClusterShutdown| - [CommitClusterStateOp] hz.laughing_blackwell.generic-operation.thread-0 - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] Changing cluster state from ClusterState{state=ACTIVE, lock=LockGuard{lockOwner=[127.0.0.1]:5061, lockOwnerId='46628225-c32e-4054-9db8-9d4c8f184db8', lockExpiryTime=1630759763261}} to ClusterStateChange{type=class com.hazelcast.cluster.ClusterState, newState=PASSIVE}, initiator: [127.0.0.1]:5061, transient: true
12:48:03,263  INFO |testClusterShutdown| - [ClusterService] hz.ecstatic_blackwell.cached.thread-3 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Sending shut down operations to all members...
12:48:03,263  INFO |testClusterShutdown| - [ShutdownNodeOp] hz.laughing_blackwell.priority-generic-operation.thread-0 - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] Shutting down node in cluster passive state. Requested by: [127.0.0.1]:5061
12:48:03,264  INFO |testClusterShutdown| - [LifecycleService] hz.laughing_blackwell..clusterShutdown - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] [127.0.0.1]:5055 is SHUTTING_DOWN
12:48:03,264  INFO |testClusterShutdown| - [Node] hz.laughing_blackwell..clusterShutdown - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] Shutting down connection manager...
12:48:03,265  INFO |testClusterShutdown| - [TcpServerConnection] hz.laughing_blackwell..clusterShutdown - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] Connection[id=3, /127.0.0.1:5055->/127.0.0.1:52114, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5061, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: TcpServer is stopping
12:48:03,266  INFO |testClusterShutdown| - [Node] hz.laughing_blackwell..clusterShutdown - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] Shutting down node engine...
12:48:03,267  WARN |testClusterShutdown| - [TcpServerConnection] hz.ecstatic_blackwell.IO.thread-in-2 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Connection[id=3, /127.0.0.1:52114->/127.0.0.1:5055, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5055, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Exception in Connection[id=3, /127.0.0.1:52114->/127.0.0.1:5055, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5055, alive=true, connectionType=MEMBER, planeIndex=0], thread=hz.ecstatic_blackwell.IO.thread-in-2
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method) ~[?:1.8.0_172]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43) ~[?:1.8.0_172]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_172]
	at sun.nio.ch.IOUtil.read(IOUtil.java:197) ~[?:1.8.0_172]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[?:1.8.0_172]
	at com.hazelcast.internal.networking.nio.NioInboundPipeline.process(NioInboundPipeline.java:114) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKey(NioThread.java:383) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKeys(NioThread.java:368) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.selectLoop(NioThread.java:294) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.executeRun(NioThread.java:249) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
12:48:03,268  INFO |testClusterShutdown| - [TcpServerConnector] hz.ecstatic_blackwell.cached.thread-5 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Connecting to /127.0.0.1:5055, timeout: 0, bind-any: true
12:48:03,269  INFO |testClusterShutdown| - [NodeExtension] hz.laughing_blackwell..clusterShutdown - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] Destroying node NodeExtension.
12:48:03,269  INFO |testClusterShutdown| - [Node] hz.laughing_blackwell..clusterShutdown - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] Hazelcast Shutdown is completed in 5 ms.
12:48:03,269  INFO |testClusterShutdown| - [LifecycleService] hz.laughing_blackwell..clusterShutdown - [127.0.0.1]:5055 [testClusterShutdown] [5.0-SNAPSHOT] [127.0.0.1]:5055 is SHUTDOWN
12:48:04,268  INFO |testClusterShutdown| - [TcpServerConnector] hz.ecstatic_blackwell.cached.thread-5 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Could not connect to: /127.0.0.1:5055. Reason: IOException[Connection refused: connect to address /127.0.0.1:5055]
12:48:05,467  INFO |testClusterShutdown| - [TcpServerConnector] hz.ecstatic_blackwell.cached.thread-2 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Connecting to /127.0.0.1:5055, timeout: 0, bind-any: true
12:48:05,752  WARN |testChangeClusterState| - [TcpServerConnection] hz.loving_blackwell.IO.thread-in-1 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] Connection[id=2, /127.0.0.1:52112->/127.0.0.1:5033, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5033, alive=false, connectionType=MEMBER, planeIndex=-1] closed. Reason: Exception in Connection[id=2, /127.0.0.1:52112->/127.0.0.1:5033, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5033, alive=true, connectionType=MEMBER, planeIndex=-1], thread=hz.loving_blackwell.IO.thread-in-1
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method) ~[?:1.8.0_172]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43) ~[?:1.8.0_172]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_172]
	at sun.nio.ch.IOUtil.read(IOUtil.java:197) ~[?:1.8.0_172]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[?:1.8.0_172]
	at com.hazelcast.internal.networking.nio.NioInboundPipeline.process(NioInboundPipeline.java:114) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKey(NioThread.java:383) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKeys(NioThread.java:368) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.selectLoop(NioThread.java:294) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.executeRun(NioThread.java:249) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
12:48:05,752  INFO |testChangeClusterState| - [TcpIpJoiner] hz.loving_blackwell.IO.thread-in-1 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5033 is added to the blacklist.
12:48:05,906  WARN |testGetClusterState| - [TcpServerConnection] hz.unruffled_blackwell.IO.thread-in-2 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] Connection[id=3, /127.0.0.1:52113->/127.0.0.1:5039, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5039, alive=false, connectionType=MEMBER, planeIndex=-1] closed. Reason: Exception in Connection[id=3, /127.0.0.1:52113->/127.0.0.1:5039, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5039, alive=true, connectionType=MEMBER, planeIndex=-1], thread=hz.unruffled_blackwell.IO.thread-in-2
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method) ~[?:1.8.0_172]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43) ~[?:1.8.0_172]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_172]
	at sun.nio.ch.IOUtil.read(IOUtil.java:197) ~[?:1.8.0_172]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[?:1.8.0_172]
	at com.hazelcast.internal.networking.nio.NioInboundPipeline.process(NioInboundPipeline.java:114) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKey(NioThread.java:383) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKeys(NioThread.java:368) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.selectLoop(NioThread.java:294) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.executeRun(NioThread.java:249) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
12:48:05,906  INFO |testGetClusterState| - [TcpIpJoiner] hz.unruffled_blackwell.IO.thread-in-2 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5039 is added to the blacklist.
12:48:06,468  INFO |testClusterShutdown| - [TcpServerConnector] hz.ecstatic_blackwell.cached.thread-2 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Could not connect to: /127.0.0.1:5055. Reason: IOException[Connection refused: connect to address /127.0.0.1:5055]
12:48:08,671  INFO |testClusterShutdown| - [TcpServerConnector] hz.ecstatic_blackwell.cached.thread-2 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Connecting to /127.0.0.1:5055, timeout: 0, bind-any: true
12:48:09,673  INFO |testClusterShutdown| - [TcpServerConnector] hz.ecstatic_blackwell.cached.thread-2 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Could not connect to: /127.0.0.1:5055. Reason: IOException[Connection refused: connect to address /127.0.0.1:5055]
12:48:09,673  WARN |testClusterShutdown| - [TcpServerConnectionErrorHandler] hz.ecstatic_blackwell.cached.thread-2 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Removing connection to endpoint [127.0.0.1]:5055 Cause => java.io.IOException {Connection refused: connect to address /127.0.0.1:5055}, Error-Count: 5
12:48:09,673  WARN |testClusterShutdown| - [MembershipManager] hz.ecstatic_blackwell.cached.thread-2 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Member [127.0.0.1]:5055 - 83c9d5a1-854d-4bc7-8f85-d202678a94be is suspected to be dead for reason: No connection
12:48:09,673  INFO |testClusterShutdown| - [MembershipManager] hz.ecstatic_blackwell.cached.thread-2 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Starting mastership claim process...
12:48:09,673  INFO |testClusterShutdown| - [MembershipManager] hz.ecstatic_blackwell.cached.thread-2 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Local MembersView{version=2, members=[MemberInfo{address=[127.0.0.1]:5055, uuid=83c9d5a1-854d-4bc7-8f85-d202678a94be, liteMember=false, memberListJoinVersion=1}, MemberInfo{address=[127.0.0.1]:5061, uuid=759052b1-9f04-46d8-9f76-204e98af6674, liteMember=false, memberListJoinVersion=2}]} with suspected members: [Member [127.0.0.1]:5055 - 83c9d5a1-854d-4bc7-8f85-d202678a94be] and initial addresses to ask: []
12:48:09,674  INFO |testClusterShutdown| - [ClusterService] hz.ecstatic_blackwell.cached.thread-5 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] 

Members {size:1, ver:3} [
	Member [127.0.0.1]:5061 - 759052b1-9f04-46d8-9f76-204e98af6674 this
]

12:48:09,674  INFO |testClusterShutdown| - [TransactionManagerService] hz.ecstatic_blackwell.cached.thread-9 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5055, UUID: 83c9d5a1-854d-4bc7-8f85-d202678a94be
12:48:09,674  INFO |testClusterShutdown| - [MembershipManager] hz.ecstatic_blackwell.cached.thread-5 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Mastership is claimed with: MembersView{version=3, members=[MemberInfo{address=[127.0.0.1]:5061, uuid=759052b1-9f04-46d8-9f76-204e98af6674, liteMember=false, memberListJoinVersion=2}]}
12:48:10,268  INFO |testClusterShutdown| - [ClusterService] hz.ecstatic_blackwell.cached.thread-3 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Number of other members remaining: 0. Shutting down itself.
12:48:10,268  INFO |testClusterShutdown| - [LifecycleService] hz.ecstatic_blackwell.cached.thread-3 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] [127.0.0.1]:5061 is SHUTTING_DOWN
12:48:10,268  INFO |testClusterShutdown| - [TextCommandServiceImpl] hz.ecstatic_blackwell.cached.thread-3 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Stopping text command service...
12:48:10,268  INFO |testClusterShutdown| - [Node] hz.ecstatic_blackwell.cached.thread-3 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Shutting down connection manager...
12:48:10,270  INFO |testClusterShutdown| - [TcpIpJoiner] hz.ecstatic_blackwell.cached.thread-7 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] [127.0.0.1]:5055 is added to the blacklist.
12:48:10,270  INFO |testClusterShutdown| - [Node] hz.ecstatic_blackwell.cached.thread-3 - [127.0.0.1]:5061 [testClusterShutdown] [5.0-SNAPSHOT] Shutting down node engine...
12:48:10,649  INFO |testChangeClusterState| - [ClusterService] testChangeClusterState - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5035 - b26361e6-8c5c-4812-b5a4-5a9f7036bcef this
]

12:48:10,649  INFO |testChangeClusterState| - [LifecycleService] testChangeClusterState - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5035 is STARTED
12:48:10,652  WARN |testChangeClusterState| - [NodeExtension] hz.reverent_blackwell.cached.thread-1 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] Security features are only available on Hazelcast Enterprise!
12:48:10,652  INFO |testChangeClusterState| - [LockClusterStateOp] hz.reverent_blackwell.cached.thread-1 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] Locking cluster state. Initiator: [127.0.0.1]:5033, lease-time: 60000
12:48:10,652  INFO |testChangeClusterState| - [LockClusterStateOp] hz.reverent_blackwell.cached.thread-1 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] Extending cluster state lock. Initiator: [127.0.0.1]:5033, lease-time: 20000
12:48:10,653  INFO |testChangeClusterState| - [CommitClusterStateOp] hz.reverent_blackwell.cached.thread-1 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] Changing cluster state from ClusterState{state=ACTIVE, lock=LockGuard{lockOwner=[127.0.0.1]:5033, lockOwnerId='1e822a53-71fa-41d3-9eab-b3788baf2361', lockExpiryTime=1630759770652}} to ClusterStateChange{type=class com.hazelcast.cluster.ClusterState, newState=FROZEN}, initiator: [127.0.0.1]:5033, transient: false
12:48:10,804  INFO |testGetClusterState| - [ClusterService] testGetClusterState - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5047 - 8957b8be-0ae4-4d20-990f-89b9bf39c7ba this
]

12:48:10,804  INFO |testGetClusterState| - [LifecycleService] testGetClusterState - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5047 is STARTED
12:48:10,804  INFO |testGetClusterState| - [LockClusterStateOp] testGetClusterState - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Locking cluster state. Initiator: [127.0.0.1]:5039, lease-time: 60000
12:48:10,804  INFO |testGetClusterState| - [LockClusterStateOp] testGetClusterState - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Extending cluster state lock. Initiator: [127.0.0.1]:5039, lease-time: 20000
12:48:10,804  INFO |testGetClusterState| - [CommitClusterStateOp] testGetClusterState - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Changing cluster state from ClusterState{state=ACTIVE, lock=LockGuard{lockOwner=[127.0.0.1]:5039, lockOwnerId='b5e58ab1-33b7-4d70-b2e9-39f0598093f9', lockExpiryTime=1630759770804}} to ClusterStateChange{type=class com.hazelcast.cluster.ClusterState, newState=FROZEN}, initiator: [127.0.0.1]:5039, transient: false
12:48:10,808  WARN |testGetClusterState| - [NodeExtension] hz.heuristic_blackwell.cached.thread-5 - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Security features are only available on Hazelcast Enterprise!
12:48:10,808  INFO |testGetClusterState| - [LockClusterStateOp] testGetClusterState - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Locking cluster state. Initiator: [127.0.0.1]:5039, lease-time: 60000
12:48:10,808  INFO |testGetClusterState| - [LockClusterStateOp] testGetClusterState - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Extending cluster state lock. Initiator: [127.0.0.1]:5039, lease-time: 20000
12:48:10,808  INFO |testGetClusterState| - [CommitClusterStateOp] testGetClusterState - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Changing cluster state from ClusterState{state=FROZEN, lock=LockGuard{lockOwner=[127.0.0.1]:5039, lockOwnerId='50790525-ad28-4bf9-af62-75d2cafe2962', lockExpiryTime=1630759770808}} to ClusterStateChange{type=class com.hazelcast.cluster.ClusterState, newState=PASSIVE}, initiator: [127.0.0.1]:5039, transient: false
12:48:10,811  WARN |testGetClusterState| - [NodeExtension] hz.unruffled_blackwell.cached.thread-6 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] Security features are only available on Hazelcast Enterprise!
12:48:11,243  INFO |testGetClusterState| - [LifecycleService] Thread-37355 - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5039 is SHUTTING_DOWN
12:48:11,243  WARN |testGetClusterState| - [Node] Thread-37355 - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Terminating forcefully...
12:48:11,243  INFO |testGetClusterState| - [TextCommandServiceImpl] Thread-37355 - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Stopping text command service...
12:48:11,243  INFO |testGetClusterState| - [Node] Thread-37355 - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Shutting down connection manager...
12:48:11,244  INFO |testGetClusterState| - [Node] Thread-37355 - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Shutting down node engine...
12:48:11,247  INFO |testGetClusterState| - [NodeExtension] Thread-37355 - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Destroying node NodeExtension.
12:48:11,248  INFO |testGetClusterState| - [Node] Thread-37355 - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] Hazelcast Shutdown is completed in 5 ms.
12:48:11,248  INFO |testGetClusterState| - [LifecycleService] Thread-37355 - [127.0.0.1]:5039 [testGetClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5039 is SHUTDOWN
12:48:11,248  INFO |testGetClusterState| - [LifecycleService] Thread-37355 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5047 is SHUTTING_DOWN
12:48:11,248  WARN |testGetClusterState| - [Node] Thread-37355 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] Terminating forcefully...
12:48:11,248  INFO |testGetClusterState| - [TextCommandServiceImpl] Thread-37355 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] Stopping text command service...
12:48:11,248  INFO |testGetClusterState| - [Node] Thread-37355 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] Shutting down connection manager...
12:48:11,249  INFO |testGetClusterState| - [Node] Thread-37355 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] Shutting down node engine...
12:48:11,252  INFO |testGetClusterState| - [NodeExtension] Thread-37355 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] Destroying node NodeExtension.
12:48:11,252  INFO |testGetClusterState| - [Node] Thread-37355 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] Hazelcast Shutdown is completed in 4 ms.
12:48:11,252  INFO |testGetClusterState| - [LifecycleService] Thread-37355 - [127.0.0.1]:5047 [testGetClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5047 is SHUTDOWN
```

Standard output for `testGetClusterState`:
```
12:50:01,639  INFO |testChangeClusterState| - [InvocationMonitor] hz.loving_blackwell.InvocationMonitorThread - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] Invocations:1 timeouts:1 backup-timeouts:0
12:50:11,091  INFO |testChangeClusterState| - [LifecycleService] Thread-37333 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5033 is SHUTTING_DOWN
12:50:11,091  WARN |testChangeClusterState| - [Node] Thread-37333 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] Terminating forcefully...
12:50:11,091  INFO |testChangeClusterState| - [TextCommandServiceImpl] Thread-37333 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] Stopping text command service...
12:50:11,092  INFO |testChangeClusterState| - [Node] Thread-37333 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] Shutting down connection manager...
12:50:11,094  INFO |testChangeClusterState| - [Node] Thread-37333 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] Shutting down node engine...
12:50:11,104  INFO |testChangeClusterState| - [NodeExtension] Thread-37333 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] Destroying node NodeExtension.
12:50:11,104  INFO |testChangeClusterState| - [Node] Thread-37333 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] Hazelcast Shutdown is completed in 13 ms.
12:50:11,104  INFO |testChangeClusterState| - [LifecycleService] Thread-37333 - [127.0.0.1]:5033 [testChangeClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5033 is SHUTDOWN
12:50:11,104  INFO |testChangeClusterState| - [LifecycleService] Thread-37333 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5035 is SHUTTING_DOWN
12:50:11,104  WARN |testChangeClusterState| - [Node] Thread-37333 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] Terminating forcefully...
12:50:11,104  INFO |testChangeClusterState| - [Node] Thread-37333 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] Shutting down connection manager...
12:50:11,107  INFO |testChangeClusterState| - [Node] Thread-37333 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] Shutting down node engine...
12:50:11,110  INFO |testChangeClusterState| - [NodeExtension] Thread-37333 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] Destroying node NodeExtension.
12:50:11,110  INFO |testChangeClusterState| - [Node] Thread-37333 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] Hazelcast Shutdown is completed in 6 ms.
12:50:11,110  INFO |testChangeClusterState| - [LifecycleService] Thread-37333 - [127.0.0.1]:5035 [testChangeClusterState] [5.0-SNAPSHOT] [127.0.0.1]:5035 is SHUTDOWN
```

--- a/hazelcast/src/main/java/com/hazelcast/internal/server/tcp/SingleProtocolDecoder.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/server/tcp/SingleProtocolDecoder.java
@@ -104,7 +104,7 @@ public class SingleProtocolDecoder
             initConnection();
             setupNextDecoder();
             if (!channel.isClientMode()) {
-                // Set up the next encoder in the pipeline if  in client mode
+                // Set up the next encoder in the pipeline if in server mode
                 // This replaces SignalProtocolEncoder with next one in the pipeline
                 encoder.setupNextEncoder();
 

--- a/hazelcast/src/main/java/com/hazelcast/internal/server/tcp/SingleProtocolDecoder.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/server/tcp/SingleProtocolDecoder.java
@@ -99,12 +99,19 @@ public class SingleProtocolDecoder
 
             verifyProtocol(loadProtocol());
             encoder.signalProtocolVerified();
-            // initialize the connection
+
+            // Initialize the connection
             initConnection();
             setupNextDecoder();
+            if (!channel.isClientMode()) {
+                // Set up the next encoder in the pipeline if  in client mode
+                // This replaces SignalProtocolEncoder with next one in the pipeline
+                encoder.setupNextEncoder();
 
-            if (shouldSignalProtocolLoaded()) {
-                ((MemberProtocolEncoder) encoder.getFirstOutboundHandler()).signalProtocolLoaded();
+                // Signal the member protocol encoder only if it's needed
+                if (shouldSignalMemberProtocolEncoder) {
+                    ((MemberProtocolEncoder) encoder.getFirstOutboundHandler()).signalProtocolLoaded();
+                }
             }
 
             return CLEAN;
@@ -114,7 +121,7 @@ public class SingleProtocolDecoder
     }
 
     protected void setupNextDecoder() {
-        // replace this handler with the next one
+        // Replace this handler with the next one
         channel.inboundPipeline().replace(this, inboundHandlers);
     }
 
@@ -144,9 +151,5 @@ public class SingleProtocolDecoder
             TcpServerConnection connection = (TcpServerConnection) channel.attributeMap().get(ServerConnection.class);
             connection.setConnectionType(ConnectionType.MEMBER);
         }
-    }
-
-    private boolean shouldSignalProtocolLoaded() {
-        return !channel.isClientMode() && shouldSignalMemberProtocolEncoder;
     }
 }

--- a/hazelcast/src/main/java/com/hazelcast/internal/server/tcp/SingleProtocolEncoder.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/server/tcp/SingleProtocolEncoder.java
@@ -74,8 +74,10 @@ public class SingleProtocolEncoder extends OutboundHandler<Void, ByteBuffer> {
                 }
             }
 
-            // Set up the next encoder in the pipeline
-            setupNextEncoder();
+            if (channel.isClientMode()) {
+                // Set up the next encoder in the pipeline if  in client mode
+                setupNextEncoder();
+            }
 
             return CLEAN;
         } finally {

--- a/hazelcast/src/main/java/com/hazelcast/internal/server/tcp/SingleProtocolEncoder.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/server/tcp/SingleProtocolEncoder.java
@@ -75,7 +75,7 @@ public class SingleProtocolEncoder extends OutboundHandler<Void, ByteBuffer> {
             }
 
             if (channel.isClientMode()) {
-                // Set up the next encoder in the pipeline if  in client mode
+                // Set up the next encoder in the pipeline if in client mode
                 setupNextEncoder();
             }
 

